.github/workflows/issue-routing-helper.yml

```diff
@@ -18,38 +18,60 @@ jobs:
       &&
       !contains(github.event.issue.labels.*.name, 'Status: In Progress')
     steps:
+      - name: "Check Membership of issue creator"
+        id: "check_membership"
+        env:
+          issue_number: ${{ github.event.issue.number }}
+        run: |
+          author_association=$(gh api "repos/$GH_REPO/issues/$issue_number" -q .author_association)
+          echo "membership=$author_association" >> $GITHUB_OUTPUT
       - name: "Ensure a single 'Team: *' label with 'Status: Untriaged'"
         env:
           team_label: ${{ github.event.label.name }}
           issue_number: ${{ github.event.issue.number }}
+          membership: ${{ steps.check_membership.outputs.membership }}
+        if: >-
+          env.membership != 'OWNER'
+          &&
+          env.membership != 'MEMBER'
         run: |
-          labels_to_remove="$(gh api --paginate "/repos/$GH_REPO/labels" | jq -r --arg label_name "$team_label" '[.[].name | select((startswith("Team: ") or startswith("Status: ")) and . != $label_name and . != "Status: Untriaged")] | join(",")')
-          gh issue edit "$issue_number" --remove-label "$labels_to_remove" --add-label "${team_label},Status: Untriaged"
+          labels_to_remove=$(gh api --paginate "/repos/$GH_REPO/labels" -q '[.[].name | select((startswith("Team: ") or startswith("Status: ")) and . != "${{ github.event.label.name }}" and . != "Status: Untriaged")] | join(",")')
```

Comment:
This file is simply not up to date with `getsentry/sentry` and `getsentry/sentry-docs` so updating that here.